app/src/main/java/pl/droidsonroids/droidsmap/feature/office/mvp/OfficePresenter.kt

```diff
@@ -3,24 +3,22 @@ package pl.droidsonroids.droidsmap.feature.office.mvp
 import pl.droidsonroids.droidsmap.base.DataObserverAdapter
 import pl.droidsonroids.droidsmap.base.Presenter
 import pl.droidsonroids.droidsmap.feature.office.business_logic.OfficeFeatureBoundary
+import pl.droidsonroids.droidsmap.feature.room.ui.RoomUiFeatureView
 import pl.droidsonroids.droidsmap.model.Coordinates
 
-class OfficePresenter private constructor(
-        private val officeFeatureBoundary: OfficeFeatureBoundary) : Presenter<OfficeMvpView>() {
+open class OfficePresenter private constructor(
+        private val officeFeatureBoundary: OfficeFeatureBoundary) : OfficePresenterContract() {
 
-    override fun onFlowNavigatorRegistered() {
-        view.initMap()
-    }
+    override fun onViewAttached() = view.initMap()
 
-    fun onRequestOffice() = officeFeatureBoundary.requestOffice(OfficeDataObserver())
+    override fun onRequestOffice() = officeFeatureBoundary.requestOffice(OfficeDataObserver())
 
-    fun onRoomClicked(coordinates: Coordinates) {
-        view.animateCameraToClickedRoom(coordinates)
-    }
+    override fun onRoomClicked(coordinates: Coordinates) = view.animateCameraToClickedRoom(coordinates)
 
-    fun onMapCameraAnimationCompleted() {
+    override fun onMapCameraAnimationCompleted() {
         view.prepareForRoomTransition()
         view.performRoomTransition()
+        flowChangeNavigator?.changePerspective(RoomUiFeatureView::class)
```

Comment:
Navigator nie będzie nullem, bo jest rejestrowany w bazowym widoku. Jeżeli zaś inny programista świadomie nadpisze metodę i nie zainicjalizuje navigatora (może się zdarzyć taki widok, że nie będzie chciał nigdzie nawigować), a potem będzie się do niego odwoływał, to już chyba nie nasza brocha. Domyślnie jest dostarczona działająca implementacja.